src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/NodeMonitoringConnectionPlugin.java

```diff
@@ -293,7 +295,15 @@ private void checkIfChanged(JdbcConnection newConnection) {
    */
   private void generateNodeKeys(Connection connection) {
     this.nodeKeys.clear();
+
     final HostInfo hostInfo = this.currentConnectionProvider.getCurrentHostInfo();
+    this.nodeKeys.add(
```

Comment:
Why is this moved up? 
Originally it was at the bottom to have the `IP & port` combo first in the set/list.

Reply:
No particular reason. Just move it closer to hostinfo. Since nodeKeys is a Set<> there's no difference in what order you add values to it.